components/get_link_modal.tsx

```diff
@@ -1,104 +1,101 @@
+        const textarea = this.textAreaRef.current;
 
-        try {
-            this.setState({copiedLink: document.execCommand('copy')});
-        } catch (err) {
-            this.setState({copiedLink: false});
-        }
-    }
+        if (textarea) {
+            textarea.focus();
+            textarea.setSelectionRange(0, this.props.link.length);
 
-    render() {
-        let helpText = null;
-        if (this.props.helpText) {
-            helpText = (
-                <p>
-                    {this.props.helpText}
-                    <br/>
-                    <br/>
-                </p>
-            );
+            try {
+                this.setState({copiedLink: document.execCommand('copy')});
+            } catch (err) {
+                this.setState({copiedLink: false});
+            }
         }
+    }
 
-        let copyLink = null;
+    public render(): JSX.Element {
+        const helpText = this.props.helpText && (
+            <p>
+                {this.props.helpText}
+                <br/>
+                <br/>
+            </p>
+        );
 
-        if (document.queryCommandSupported('copy')) {
-            copyLink = (
-                <button
-                    id='linkModalCopyLink'
-                    data-copy-btn='true'
-                    type='button'
-                    className='btn btn-primary pull-left'
-                    onClick={this.copyLink}
-                >
-                    <FormattedMessage
-                        id='get_link.copy'
-                        defaultMessage='Copy Link'
-                    />
-                </button>
-            );
-        }
+        const copyLink = document.queryCommandSupported('copy') && (
+            <button
+                id='linkModalCopyLink'
+                data-copy-btn='true'
+                type='button'
+                className='btn btn-primary pull-left'
+                onClick={this.copyLink}
+            >
+                <FormattedMessage
+                    id='get_link.copy'
+                    defaultMessage='Copy Link'
+                />
+            </button>
+        );
 
         const linkText = (
             <textarea
                 id='linkModalTextArea'
                 className='form-control no-resize min-height'
-                ref='textarea'
+                ref={this.textAreaRef}
                 value={this.props.link}
                 onClick={this.copyLink}
                 readOnly={true}
             />
         );
 
-        let copyLinkConfirm = null;
-        if (this.state.copiedLink) {
-            copyLinkConfirm = (
-                <p className='alert alert-success alert--confirm'>
-                    <SuccessIcon/>
-                    <FormattedMessage
-                        id='get_link.clipboard'
-                        defaultMessage=' Link copied'
-                    />
-                </p>
-            );
-        }
+        const copyLinkConfirm = this.state.copiedLink && (
```

Comment:
Is this change necessary?  If not, please revert to original form.  Same case for `copyLink` and `helpText` above.

Reply:
@Phizzard If you don't mind, I'll push commit per requested change.